rpyc/rpyc_client/command_handlers.py

```diff
@@ -0,0 +1,203 @@
+            except (psutil.NoSuchProcess, psutil.AccessDenied, psutil.ZombieProcess):
+                pass
+
+        return process_list, None
+
+
+class FileStatHandler(CommandHandler):
+    def _parse_input(self, split_input, additional_input):
+        if len(split_input) != 2:
+            raise TypeError('expecting one argument')
+        return {'path': split_input[1]}
+
+    def _execute(self, path):
+        remote_os = self.rpyc_conn.modules.os
+        return remote_os.stat(path), None
+
+
+class KillProcessHandler(CommandHandler):
+    def _parse_input(self, split_input, additional_input):
+        if len(split_input) < 2:
+            raise TypeError('expects at least one argument')
+
+        if len(split_input) == 3:
+            if not split_input[1].startswith('-'):
+                raise ValueError("kill signal number should start with '-'")
+            return {'signal': int(split_input[1][1:]), 'pid': int(split_input[2])}
+        elif len(split_input) == 2:
+            return {'pid': int(split_input[1])}
+
+        raise TypeError('too many arguments')
+
+    def _execute(self, pid, signal=9):
+        remote_os = self.rpyc_conn.modules.os
+        remote_os.kill(pid, signal)
+        return self.empty_output
+
+
+class RunAsNewProcessHandler(CommandHandler):
+    def _parse_input(self, split_input, additional_input):
+        if len(split_input) < 2:
+            raise TypeError('expecting at least one argument')
+        return {'path_to_executable': split_input[1], 'argv_to_exe': split_input[2:]}
+
+    def _execute(self, path_to_executable, argv_to_exe):
+        remote_subprocess = self.rpyc_conn.modules.subprocess
+        string_args = ' '.join(argv_to_exe)
+        command_line_input = path_to_executable + ' ' + string_args
+        process = remote_subprocess.Popen(command_line_input)
+        return None, process
+
+
+class MonitorHandler(CommandHandler):
+    def _parse_input(self, split_input, additional_input):
+        if len(split_input) != 3:
+            raise TypeError("expecting 2 arguments")
+
+        if split_input[1] == '-r':
+            return {'monitored_path': split_input[2], 'log_path': '', 'monitors': additional_input, 'remove': True}
+
+        return {'monitored_path': split_input[1], 'log_path': split_input[2], 'monitors': additional_input}
+
+    def _execute(self, monitored_path, log_path, monitors, remove=False):
+        if not remove and monitored_path in monitors:
+            return self.empty_output
+
+        if remove and monitored_path in monitors:
+            monitors[monitored_path].stop()
+            del monitors[monitored_path]
+            return self.empty_output
+
+        if remove and monitored_path not in monitors:
+            return self.empty_output
+
+        # not remove and monitored_path not in monitors
+        return '', self.rpyc_conn.root.FileMonitor(monitored_path, self.__get_callback(log_path))
+
+    # Warning: Careful with this closure
```

Comment:
?

Reply:
They said that sometimes closures doesn't work well with rpyc. But this works fine so I'll remove this comment.